spring-cloud-deployer-spi/src/main/java/org/springframework/cloud/deployer/spi/app/observation/DefaultAppDeployerObservationConvention.java

```diff
@@ -0,0 +1,94 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.deployer.spi.app.observation;
+
+import java.util.Map;
+
+import io.micrometer.common.KeyValues;
+
+import org.springframework.cloud.deployer.spi.core.AppDeploymentRequest;
+import org.springframework.cloud.deployer.spi.core.RuntimeEnvironmentInfo;
+import org.springframework.lang.Nullable;
+import org.springframework.util.StringUtils;
+
+public class DefaultAppDeployerObservationConvention implements AppDeployerObservationConvention {
+
+	// TODO: Check what is high and what is low cardinality
+	@Override
+	public KeyValues getLowCardinalityKeyValues(AppDeployerContext context) {
+		KeyValues keyValues = KeyValues.empty();
+		if (context.getAppId() != null) {
+			keyValues = keyValues.and(AppDeployerDocumentedObservation.Tags.APP_ID.withValue(context.getAppId()));
+		}
+		if (context.getAppScaleRequest() != null) {
+			keyValues = keyValues.and(AppDeployerDocumentedObservation.ScaleTags.DEPLOYER_SCALE_DEPLOYMENT_ID.withValue(context.getAppScaleRequest().getDeploymentId()),
+					AppDeployerDocumentedObservation.ScaleTags.DEPLOYER_SCALE_COUNT.withValue(String.valueOf(context.getAppScaleRequest().getCount())));
+		}
+		if (context.getRuntimeEnvironmentInfo() != null) {
+			keyValues = keyValues.and(platformSpecificKeyValues(context.getRuntimeEnvironmentInfo(), context.getRequest()));
+		}
+		return keyValues;
+	}
+
+
+	private KeyValues platformSpecificKeyValues(RuntimeEnvironmentInfo environmentInfo, @Nullable AppDeploymentRequest request) {
+		KeyValues keyValues = KeyValues.empty();
+		Map<String, String> platformSpecificInfo = environmentInfo.getPlatformSpecificInfo();
+		if (request != null) {
+			String platformName = request.getDeploymentProperties().get("spring.cloud.deployer.platformName");
+			if (StringUtils.hasText(platformName)) {
+				keyValues = keyValues.and(AppDeployerDocumentedObservation.Tags.PLATFORM_NAME.withValue(platformName));
+			}
+			String appName = request.getDeploymentProperties().get("spring.cloud.deployer.appName");
+			if (StringUtils.hasText(appName)) {
+				keyValues = keyValues.and(AppDeployerDocumentedObservation.Tags.APP_NAME.withValue(appName));
+			}
+			String group = request.getDeploymentProperties().get("spring.cloud.deployer.group");
+			if (StringUtils.hasText(group)) {
+				keyValues = keyValues.and(AppDeployerDocumentedObservation.Tags.APP_GROUP.withValue(group));
+			}
+		}
+		keyValues = keyValues.and(addCfTags(platformSpecificInfo));
+		return keyValues.and(addK8sTags(platformSpecificInfo));
+	}
+
+	private KeyValues addCfTags(Map<String, String> platformSpecificInfo) {
+		KeyValues keyValues = KeyValues.empty();
+		if (platformSpecificInfo.containsKey("API Endpoint")) {
```

Comment:
Where do these string constants come from? Same `?` for the K8S ones below.

Reply:
I think I looked at the Kubernetes and CF deployers. So this instrumentation is taken from Sleuth where I had to assume everything there. Now with Spring Observability what we can do is move the platform specific code to the corresponding deployers. Then all the platform specific tagging would be removed from this class. WDYT?

Reply:
I agree.  This also covers my comment about Local deployer as well as the condition when we add a new deployer for another platform.

Reply:
I'll leave it for now here so that we agree to the principles and then in `kubernetes` or `cf` deployers we will just add global conventions that will set the key values

Reply:
Sounds great. Should we add a TODO in here, create an issue for this, OR just sit back and relax and know that @marcingrzejszczak will made this change? ;). I'm fine w/ any of these reminder mechanisms.